loopy/codegen/__init__.py

```diff
@@ -190,7 +190,7 @@ class CodeGenerationState:
     seen_functions: Set[SeenFunction]
     seen_atomic_dtypes: Set[LoopyType]
 
-    var_subst_map: Map[str, ExpressionT]
+    var_subst_map: constantdict[str, ExpressionT]
```

Comment:
Whenever possible, document as `Mapping`.

Reply:
As far as I can see, I annotated as `Mapping` in all other places. In this location, I think it can't be done, since `var_subst_map` is called with methods that aren't part of `Mapping` below:

- `var_subst_map.set()`, `var_subst_map.update()`

`
loopy/codegen/__init__.py:219: error: "Mapping[str, Union[Union[int, signedinteger[_8Bit], signedinteger[_16Bit], signedinteger[_32Bit], signedinteger[_64Bit], unsignedinteger[_8Bit], unsignedinteger[_16Bit], unsignedinteger[_32Bit], unsignedinteger[_64Bit]], Union[float, complex, floating[_32Bit], floating[_64Bit], complexfloating[_32Bit, _32Bit], complexfloating[_64Bit, _64Bit]], Any]]" has no attribute "set"  [attr-defined]
`

Reply:
OK.